src/main/java/com/iexec/resultproxy/jwt/JwtService.java

```diff
@@ -83,21 +83,31 @@ private byte[] initKey(String jwtKeyPath) throws IOException {
      * <p>
      * A new signed token will be created in the following cases:
      * <ul>
-     * <li> If a JWT is not found, an exception will be thrown and a new token will be created.
+     * <li> If a JWT is not found in storage for the given wallet address.
      * <li> If a JWT is found but was not signed with the correct key.
      * @param walletAddress Ethereum address for which
      * @return A valid JWT token signed with this instance key.
      */
     private String getOrCreateJwt(String walletAddress) {
+        Optional<Jwt> oJwt = findByWalletAddress(walletAddress);
+        if (oJwt.isEmpty()) {
+            log.info("JWT token does not exist in storage for {}, generating a new one", walletAddress);
+            Jwt jwt = new Jwt(walletAddress, createJwt(walletAddress));
+            save(jwt);
+            return jwt.getJwtString();
+        }
         String jwtString;
         try {
-            Jwt jwt = findByWalletAddress(walletAddress).orElseThrow();
-            jwtString = jwt.getJwtString();
+            jwtString = oJwt.get().getJwtString();
             getWalletAddressFromJwtString(jwtString);
-        } catch (IllegalArgumentException | JwtException | NoSuchElementException e) {
-            log.warn("Valid JWT token not found in storage, generating a new one");
+            log.info("Valid token found for {}", walletAddress);
+        } catch (IllegalArgumentException | JwtException e) {
+            log.warn("Valid JWT token not found in storage for {}, generating a new one", walletAddress);
             jwtString = createJwt(walletAddress);
-            save(new Jwt(walletAddress, jwtString));
+            // Update existing JWT to keep ID
+            Jwt jwt = oJwt.get();
+            jwt.setJwtString(jwtString);
+            save(jwt);
```

Comment:
```
    private String getOrCreateJwt(String walletAddress) {
        Jwt jwt;
        Optional<Jwt> foundJwt = findByWalletAddress(walletAddress);
        if(foundJwt.isPresent()){
            jwt = foundJwt.get();
            if(isJwtStillValid(walletAddress, jwt)){
                return jwt.getJwtString();
            }
            log.warn("JWT token found is now invalid {}, generating a new one", walletAddress);
        } else {
            log.info("JWT token does not exist in storage for {}, generating a new one", walletAddress);
            jwt = new Jwt();
            jwt.setWalletAddress(walletAddress);
        }
        jwt.setJwtString(createJwt(walletAddress));
        save(jwt);
        return jwt.getJwtString();
    }
```

Reply:
Rework done in 2fee8b9.
I removed the Optional and went for something more similar to the starting point.